hlx_statics/scripts/lib-helix.js

```diff
@@ -730,6 +731,44 @@ export async function loadSection(section, loadCallback) {
   }
 }
 
+/**
+ *  Add block attributes from MD to the respective block.
+ */
+
+function addBlockAttributes(){
+  const allBlocks = document.querySelectorAll('.block');
+  allBlocks.forEach(block => {
+    const strongTags = block.querySelectorAll('strong');
+    strongTags.forEach(strongTag => {
+      const dataContent = strongTag.textContent.trim();
+      if (dataContent.startsWith('data-')) {
+        const keyValuePairs = [];
+        let currentPair = '';
+        let insideParentheses = false;
+        for (let i = 0; i < dataContent.length; i++) {
+          const char = dataContent[i];
+          if (char === '(') insideParentheses = true;
+          else if (char === ')') insideParentheses = false;
+          if (!insideParentheses && char === ',') {
```

Comment:
This logic of iterating through each character is a bit inefficient (because we're creating a string each time), and may not work for edge cases where attribute values contain commas or nested parentheses.

I think rather than adding all attributes in one row, we should add each attribute on their own row. This would make parsing each key-value pair less error-prone.

Also currently we have two slightly different ways of handling attributes:
1. one attribute per row: https://github.com/aemsites/devsite-runtime-connector/blob/main/src/steps/mdx-to-gridtables.ts#L80
2. all attributes in one row: https://github.com/aemsites/devsite-runtime-connector/blob/main/src/steps/mdx-to-gridtables.ts#L120

They're currently a redundant for RedoclyAPIBlock (see first vs. subsequent rows): https://one-repo--adp-devsite--adobedocs.aem.page/github-actions-test/test/redocly-api-block.md

I think it would be nice to consolidate them into one to reduce dupe code.  I currently have the parsing of one-attribute-per-row implemented in a PR here https://github.com/AdobeDocs/adp-devsite/pull/22/files#diff-36a162357d2bbb29af1040f0270dbccd1d78339c71277a8b6df01b1ae5d9a8e3R31-R47. And all-attributes-in-one-row here https://github.com/AdobeDocs/adp-devsite/pull/22/files#diff-36a162357d2bbb29af1040f0270dbccd1d78339c71277a8b6df01b1ae5d9a8e3R49-R54. If we consolidate the two approaches, we might be to simplify that PR's `redoclyapiblock.js` and this PR's `lib-helix.js`.

I know this'll mean we revisit the drawing board a bit, but I think it's worth it. @timkim @manimitrah let me know what you think

Reply:
Yes, @melissag-ensemble, I have updated the parse method. Kindly review it.